core/templates/pages/exploration-editor-page/feedback-tab/services/thread-data-backend-api.service.ts

```diff
@@ -130,9 +130,12 @@ export class ThreadDataBackendApiService {
     return thread;
   }
 
+  // 'suggestionBackendDict' will be undefined when threadId is invalid
+  // resulting the function to throw an error.
```

Comment:
What about the `threadBackendDict`?

Reply:
![image](https://user-images.githubusercontent.com/60519303/128304814-f7fe4dd0-e5fd-4f11-acc6-a0367f4a2957.png)
`.get` here can return undefined if dict is null which makes suggestionBackendDict to be of type `SuggestionBackendDict | undefined` this is not the case with `threadBackendDict1

Reply:
done